homeassistant/components/zha/core/device.py

```diff
@@ -0,0 +1,336 @@
+                _LOGGER.warning(
+                    '%s errored while configuring listener: %s ex: %s',
+                    self.name,
+                    "{}-{}".format(listener.name, listener.unique_id),
+                    ex
+                )
+        _LOGGER.debug('%s: completed configuration.', self.name)
+
+    async def async_initialize(self, from_cache):
+        """Initialize listeners."""
+        _LOGGER.debug('%s: started initialization.', self.name)
+        for listener in self.all_listeners:
+            try:
+                await listener.async_initialize(from_cache)
+                _LOGGER.debug('%s: listener: %s initialized - from cache: %s.',
+                              self.name,
+                              "{}-{}".format(
+                                  listener.name, listener.unique_id),
+                              from_cache)
+            except Exception as ex:  # pylint: disable=broad-except
+                _LOGGER.warning(
+                    '%s errored while initializing listener: %s ex: %s',
+                    self.name,
+                    "{}-{}".format(listener.name, listener.unique_id),
+                    ex
+                )
+        _LOGGER.debug('%s: completed initialization.', self.name)
+
+    async def async_accept_messages(self):
+        """Start accepting messages from the zigbee network."""
+        for listener in self.all_listeners:
+            await listener.accept_messages()
+            _LOGGER.debug('%s: listener: %s is now accepting messages.',
+                          self.name, "{}-{}".format(
+                              listener.name, listener.unique_id))
+
+    async def async_unsub_dispatcher(self):
+        """Unsubscribe the dispatcher."""
+        if self._unsub:
+            self._unsub()
+
+    async def get_clusters(self):
+        """Get all clusters for this device."""
+        clusters = {}
+        for ep_id, endpoint in self._zigpy_device.endpoints.items():
+            if ep_id == 0:
+                continue
+            clusters[ep_id] = {
+                IN: endpoint.in_clusters,
+                OUT: endpoint.out_clusters
+            }
+        return clusters
+
+    async def get_cluster(self, endpooint_id, cluster_id, cluster_type=IN):
+        """Get zigbee cluster from this entity."""
+        clusters = await self.get_clusters()
+        if cluster_type == IN:
+            cluster = clusters[endpooint_id][IN][cluster_id]
+        else:
+            cluster = clusters[endpooint_id][OUT][cluster_id]
+        return cluster
+
+    async def get_cluster_attributes(self, endpooint_id, cluster_id,
+                                     cluster_type=IN):
+        """Get zigbee attributes for specified cluster."""
+        cluster = await self.get_cluster(endpooint_id, cluster_id,
+                                         cluster_type)
+        if cluster is None:
+            return
```

Comment:
This is an inconsistent return. We should `return None` explicitly if we return something anywhere.